### DISCUSSION TITLE
Installation error: /usr/lib64/libOpenGL.so: undefined reference to `_glapi_tls_Current'

### DISCUSSION THREAD

**Discussion by anshuhim20** [ANTsX/ANTs#1371] [General]
Dear All,
I have used the following link for installation  of ANTs on my RHEL Red Hat Enterprise Linux Workstation release 7.9 using https://sourceforge.net/p/advants/discussion/840261/thread/53dffda1/#619a/f05e. I have NVIDIA graphics card installed and the output of $ rpm -qa *\nvidia\* is :
nvidia-kmod-387.26-2.el7.x86_64
xorg-x11-drv-nvidia-libs-387.26-1.el7.x86_64
xorg-x11-drv-nvidia-devel-387.26-1.el7.x86_64
xorg-x11-drv-nvidia-387.26-1.el7.x86_64
xorg-x11-drv-nvidia-gl-387.26-1.el7.x86_64 

I am also attaching the build log as well
[build.log](https://github.com/ANTsX/ANTs/files/8829753/build.log)

I have also tried putting cmake -DWITH_OPENGL=OFF -S /home/USER/ANTs -B /home/USER/build/VTK in my build directory as suggested here https://askubuntu.com/questions/1197026/libgl-so-undefined-reference-to-glapi-tls-current it executes though with the below message 
Not using CCache
-- SuperBuild - ANTS => Requires ITKv5, 
-- SuperBuild -   ITKv5[OK]
-- SuperBuild - ANTS[OK]
-- Configuring done
-- Generating done
-- Build files have been written to: /home/DST-CSRI/build/VTK

And after compiling as per the above mentioned link ccmake ~/ANTs
the error is given below
[ 56%] Linking CXX executable ../../bin/vtkProbeOpenGLVersion-9.1
/usr/lib64/libOpenGL.so: undefined reference to `_glapi_tls_Current'
collect2: error: ld returned 1 exit status
make[5]: *** [bin/vtkProbeOpenGLVersion-9.1] Error 1
make[4]: *** [Rendering/OpenGL2/CMakeFiles/vtkProbeOpenGLVersion.dir/all] Error 2
make[3]: *** [all] Error 2
make[2]: *** [VTK-prefix/src/VTK-stamp/VTK-install] Error 2
make[1]: *** [CMakeFiles/VTK.dir/all] Error 2
make: *** [all] Error 2

Kindly rescue

Regards
Himanshu

**Comment by dorianps** [comment]
Looks like you are building VTK too. Is that intentional, do you need that?

I have a similar system that I am working to fix a few things. If you can't find the solution, I can try a fresh ANTs install in a few days.

**Comment by anshuhim20** [comment]
Thank you,
Can you help me building VTK. currently facing the attached issue.
I am using Redhat 7 operating system; trying to install ANT, which requires
building of VTK. Below is the process I have performed the following
commands:

$ git clone https://github.com/ANTsX/ANT
$ cd ANTs/
$ mkdir bin
$ cd bin
$ ccmake ../
Then configuring ANTs using parameters as shown below

   -
      - BUILD_TESTING: OFF
      - CMAKE_INSTALL_PREFIX: ./
      - USE_VTK: ON
      -  USE_SYSTEM_VTK: OFF
      After configuring when I do "make -j 1"
   -
   - The error is shown below and screenshot attached:
   -

Performing download step (git clone) for 'ITKv4'
[  4%] Performing download step (git clone) for 'ITKv4'
fatal: unable to connect to github.com:
github.com[0: 20.207.73.82]: errno=Connection timed out

Cloning into 'ITKv4'...
fatal: unable to connect to github.com:
github.com[0: 20.207.73.82]: errno=Connection timed out

Cloning into 'ITKv4'...
fatal: unable to connect to github.com:
github.com[0: 20.207.73.82]: errno=Connection timed out

Cloning into 'ITKv4'...
-- Had to git clone more than once:
          3 times.
CMake Error at ITKv4-prefix/tmp/ITKv4-gitclone.cmake:40 (message):
  Failed to clone repository:
  'git://github.com/InsightSoftwareConsortium/ITK.git'


make[2]: *** [ITKv4-prefix/src/ITKv4-stamp/ITKv4-download] Error 1
make[1]: *** [CMakeFiles/ITKv4.dir/all] Error 2
make: *** [all] Error 2

*After failing to reach github 3 times the command fails to clone the
repository. I am able to reproduce the same error in another workstation as
well. Fronm the above details it is evident that this problem is happening
due to a ping drop @ 20.207.73.82.*


*Kindly suggest the needful*

*Thanks and regards*

On Thu, Jul 14, 2022 at 11:43 PM dorianps ***@***.***> wrote:

> Looks like you are building VTK too. Is that intentional, do you need that?
>
> I have a similar system that I am working to fix a few things. If you
> can't find the solution, I can try a fresh ANTs install in a few days.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ANTsX/ANTs/discussions/1371#discussioncomment-3149402>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFSFQ6NRXU4IVL4BZJZUC4DVUBKC3ANCNFSM5XXMUHCA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


Himanshu Joshi
